### PR TITLE
First push with Weather API request and basic information showing

### PR DESCRIPTION
**Pull request by CalexisGonzalez**
First iteration with basic information showing of the weather api call. 
TODO -Styling and expanding functionalitie of the weather cardview.

### CODE REVIEW

File: app/src/main/java/com/example/fittracker/ConstantUtils.java [hunk 1]
```diff
@@ -19,4 +19,14 @@
     public static final String DATAHANDLER_TYPE = "text/plain";
     public static final Character COMA = ',';
     public static final String MAIL_HOST = "smtp.gmail.com";
+    public static final String WEATHER_BASE_URL = "https://api.openweathermap.org";
+    public static final int LOCATION_REQUEST_CODE = 1;
+    public static final String MARVEL_API_KEY_PUBLIC = "08cd7f2cd10462f4d20ab91d76ecec33";
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/ConstantUtils.java:22]: why do you have the MAVEL api keys?

File: app/src/main/java/com/example/fittracker/activity/MainScreenActivity.java [hunk 2]
```diff
@@ -41,4 +46,18 @@ public void onBackPressed() {
     public void onLogOutPressed() {
         presenter.onLogOutPressed();
     }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        switch (requestCode) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/activity/MainScreenActivity.java]: pass all this logic to the presenter pls

File: app/src/main/java/com/example/fittracker/mvp/model/MainScreenModel.java [hunk 3]
```diff
@@ -1,9 +1,39 @@
...
+import android.location.Location;
+
 import com.example.fittracker.mvp.contract.MainScreenContract;
+import com.example.fittracker.services.WeatherService;
 
 public class MainScreenModel implements MainScreenContract.Model {
+    private Location location;
+    private WeatherService service;
+
     public MainScreenModel() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/mvp/model/MainScreenModel.java]: here you must have the instance of the WeatherService.

File: app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java [hunk 4]
```diff
@@ -25,4 +40,43 @@ public void onBackPressed() {
...
+    @SuppressLint("MissingPermission")
+    @Override
+    public void setLocationObject() {
+        model.setLocationObject(locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER));
+    }
+
+    public void init() {
+        locationManager = view.getLocationManager();
+        checkLocationPermission();
+        model.setWeatherService(WeatherGenerator.createService(WeatherService.class));
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java]: you can pass this as a parameter in the model constructor from the activity

File: app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java [hunk 5]
```diff
@@ -25,4 +40,43 @@ public void onBackPressed() {
...
+        call = model.getService().getData(model.getLatitud(), model.getLongitude(),
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java]: you may call model.getWeatherDataFromService(and pass here all parameters that you need) and in the model you will return getService().getData(with all the parameters)

File: app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java [hunk 6]
```diff
@@ -25,4 +40,43 @@ public void onBackPressed() {
...
+                ConstantUtils.WEATHER_API_KEY, ConstantUtils.WEATHER_UNITS);
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java]: the WEATHER_API_KEY parameter is a model issue, not a presenter.. the model is the one that need to know this

File: app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java (hunk 4 above)
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java]: please separate the weather call in a separate method and put a loader so the user will know that something is happening. Then, when the info will be setted you hide the loader

File: app/src/main/java/com/example/fittracker/ConstantUtils.java [hunk 7]
```diff
@@ -19,4 +19,14 @@
     public static final String DATAHANDLER_TYPE = "text/plain";
     public static final Character COMA = ',';
     public static final String MAIL_HOST = "smtp.gmail.com";
+    public static final String WEATHER_BASE_URL = "https://api.openweathermap.org";
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/ConstantUtils.java]: put this in a global variable in gradle pls

File: app/src/main/java/com/example/fittracker/ConstantUtils.java [hunk 8]
```diff
@@ -19,4 +19,14 @@
     public static final String DATAHANDLER_TYPE = "text/plain";
     public static final Character COMA = ',';
     public static final String MAIL_HOST = "smtp.gmail.com";
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/ConstantUtils.java]: put this in a global variable in gradle pls

File: app/src/main/java/com/example/fittracker/ConstantUtils.java [hunk 9]
```diff
@@ -19,4 +19,14 @@
...
+    public static final String WEATHER_BASE_URL = "https://api.openweathermap.org";
+    public static final int LOCATION_REQUEST_CODE = 1;
+    public static final String MARVEL_API_KEY_PUBLIC = "08cd7f2cd10462f4d20ab91d76ecec33";
+    public static final String MARVEL_API_KEY_PRIVATE = "82460263f65b53b64520b5a4fc43754a77cc1d85";
+    public static final String WEATHER_API_KEY = "852c6dcc0bd878109bd0fc7bb1566c3e";
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/ConstantUtils.java]: put this in a global variable in gradle pls

File: app/src/main/java/com/example/fittracker/ConstantUtils.java [hunk 10]
```diff
@@ -19,4 +19,14 @@
...
+    public static final String WEATHER_UNITS = "metric";
+    public static final String WEATHER_ICON_URL = "https://openweathermap.org/img/w/";
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/ConstantUtils.java]: put this in a global variable in gradle pls

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/fittracker/ConstantUtils.java [hunk 11]
```diff
@@ -19,4 +19,14 @@
...
+    public static final String PNG_EXTENSION = ".png";
+    public static final String CELSIUS_EXTENSIONS = "ºC";
+    public static final int WEATHER_ICON_DIMENSION = 100;
 }
```
Comment (nicolbidone) [inline app/src/main/java/com/example/fittracker/ConstantUtils.java:28]: You can declare constants without a class doing
const val [const_id] = [const_value]

**Review by priettt — APPROVED** [review]

File: app/src/main/java/com/example/fittracker/services/Weather.java [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+    public String getIcon() {
+        return icon;
+    }
+
+    public void setIcon(String icon) {
+        this.icon = icon;
+    }
+
+}
```
Comment (nicolbidone) [inline app/src/main/java/com/example/fittracker/services/Weather.java:53]: you can fix this 
https://stackoverflow.com/questions/24093609/how-can-i-configure-android-studio-to-add-new-line-at-end-of-file

**Review by nicolbidone — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/fittracker/activity/MainScreenActivity.java (hunk 2 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/activity/MainScreenActivity.java]: Done

File: app/src/main/java/com/example/fittracker/mvp/model/MainScreenModel.java (hunk 3 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/model/MainScreenModel.java]: Done

File: app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java (hunk 4 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java]: Done

File: app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java (hunk 5 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java]: Done

File: app/src/main/java/com/example/fittracker/services/Weather.java (hunk 12 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/services/Weather.java:53]: Done

File: app/src/main/java/com/example/fittracker/ConstantUtils.java (hunk 7 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/ConstantUtils.java]: Done

File: app/src/main/java/com/example/fittracker/ConstantUtils.java (hunk 8 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/ConstantUtils.java]: Done

File: app/src/main/java/com/example/fittracker/ConstantUtils.java (hunk 9 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/ConstantUtils.java]: Done

File: app/src/main/java/com/example/fittracker/ConstantUtils.java (hunk 10 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/ConstantUtils.java]: Done

File: app/src/main/java/com/example/fittracker/ConstantUtils.java (hunk 11 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/ConstantUtils.java:28]: Done

**Review by nicolbidone — APPROVED** [review]

File: app/src/main/java/com/example/fittracker/ConstantUtils.java (hunk 1 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/ConstantUtils.java:22]: They must be used in the following exercise in the PPS plan

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java [hunk 13]
```diff
@@ -25,4 +42,74 @@ public void onBackPressed() {
...
+            @Override
+            public void onFailure(Call<WeatherPojo> call, Throwable t) {
+                view.fetchingError();
+            }
+        });
+    }
+
+    @Override
+    public void onRequestPermissionResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        switch (requestCode) {
```
Comment (priettt) [inline app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java]: Unnecesary switch case, please merge with the if sentence below

File: app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java [hunk 14]
```diff
@@ -25,4 +42,74 @@ public void onBackPressed() {
...
+                } else {
+                    checkLocationPermission();
+                }
+            }
+        }
+    }
+
+    @Override
+    public void setWeatherIcon(String url) {
+        Picasso.get().load(BuildConfig.weather_icon_url + url + ConstantUtils.PNG_EXTENSION).
```
Comment (priettt) [inline app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java]: Don't end lines with points. 

Picasso.get().
load()

should be instead: 

Picasso.get()
.load()

File: app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java (hunk 13 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java]: Done

File: app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java (hunk 14 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java]: Done

**Pull request merged**